features/functions.feature

```diff
@@ -0,0 +1,21 @@
+Feature: Functions
+  As a systems administrator
+  In order to make the configuration files more manageable
+  I want to be able to use functions to generate the final configuration.
+
+  Scenario: Generate firewall rules using an SRV record
+    A single SRV record can store multiple combinations of host+port for a
+    single service.
+
+    Given a file named "network.puffy" with:
+    """
+    node 'example.com' do
+      pass out proto tcp from any to srv('_x-puppet._tcp.blogreen.org')
```

Comment:
I am still not 100 % satisfied given the proto is part of the service (`_tpc` segment above) and repeating it is redundant and error-prone.

The idea was to have a construct that does not look-like the rest of the configuration syntax (a function-like syntax), but maybe we can think about something that does not have the above drawback?

Reply:
> `pass out to srv('_ldap._tcp.example.com')` becomes `pass out proto tcp to 127.0.0.1 port 389`.
This syntax looks good and according to implementation explanation, it seems to bring ease and flexibility, without repeats… :+1:

Reply:
Fixed in the last commit.